tests/waku_archive/test_retention_policy.nim

```diff
@@ -53,6 +54,41 @@ suite "Waku Archive - Retention policy":
 
     ## Cleanup
     (waitFor driver.close()).expect("driver to close")
+  
+  test "size retention policy - windowed message deletion":
+    ## Given
+    let
+      # in megabytes
+      sizeLimit:float = 0.05
+      excess = 123
+
+    let driver = newTestArchiveDriver()
+
+    let retentionPolicy: RetentionPolicy = SizeRetentionPolicy.init(size=sizeLimit)
+
+    require (waitFor retentionPolicy.execute(driver)).isOk()
+    
+    ## When
+    for i in 1..excess:
+      let msg = fakeWakuMessage(payload= @[byte i], contentTopic=DefaultContentTopic, ts=Timestamp(i))
+
+      require (waitFor driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp)).isOk()
```

Comment:
Would be faster (maybe?) to collect them and wait for all of them at the same time.

```
var futures = collect(newSeq):
      for i in 1..excess:
        driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp)

let finished = await allFinished(futures)
```